gui/default/syncthing/core/treeViewDirective.js

```diff
@@ -0,0 +1,27 @@
+angular.module('syncthing.core')
+    .directive('treeElement', function () {
+        return {
+            restrict: 'EA',
+            scope: {
+                fileType: "=",
+                name: "@",
+                show: "&",
+            },
+            template: `
+                <div ng-if="fileType === 0">
+                    <span style="margin:10px;" class="fas fa-file"></span>
+                    <span>{{name}}</span>
+                </div>
+                <div ng-if="fileType === 1">
+                    <a href="#" ng-click="show()">
+                        <span style="margin:10px;" class="fas fa-folder"></span>
+                        <span>{{name}}</span>
+                    </a>
+                </div>
+                <div ng-if="fileType === 4">
```

Comment:
As discussed, this should be strings, integers for enums for public APIs is a big pain.